### PR TITLE
Don't complete values in type position

### PR DESCRIPTION
**Pull request by Veykril**
Will add some proper tests in a bit

### CODE REVIEW

File: crates/ide_completion/src/context.rs [hunk 1]
```diff
@@ -29,20 +29,25 @@ pub(crate) enum PatternRefutability {
     Irrefutable,
 }
 
+#[derive(Debug)]
+pub(super) enum PathKind {
+    Expr,
```
Comment (flodiebold) [inline crates/ide_completion/src/context.rs:34]: `Value` might be a better name, since this includes patterns?

Comment (Veykril) [reply crates/ide_completion/src/context.rs:34]: This doesn't include patterns as it checks for `PathExpr`, which cannot appear in pattern position. Patterns will cause the `kind` field to just fall back to `None` for the time being.

Comment (flodiebold) [reply crates/ide_completion/src/context.rs:34]: Well ok maybe it _should_ include patterns? :smile:

Comment (Veykril) [reply crates/ide_completion/src/context.rs:34]: It *could* but there is no use for it *yet* so I don't think there is point in adding it yet. I'll tackle pattern things again next probably so I'll see if there is a use for it then.

**Pull request merged**